### DISCUSSION TITLE
Next.js 13 Exclude Some Page Layout

### DISCUSSION THREAD

**Discussion by hamedyosefian** [vercel/next.js#42227] [Help] [answered]
Hi .. The new structure nextjs we can create layout shared all page into folder ... but i want login page do not have layout ...
how exclude some page layout ?

**Comment by icyJoseph** [comment, marked as answer]
I think you need to use groups, like they document here: https://beta.nextjs.org/docs/routing/defining-routes#example-creating-multiple-root-layouts

So for example `app/(users)/layout.tsx` and  `app/(everyone)/layout.tsx` would be applied to their sub folders.

- `app/(users)/dashboard/page.tsx` would receive the layout defined under `(users)`.
- `app/(everyone)/login/page.tsx` would receive the layout defined under `(everyone)`.
- `app/(everyone)/page.tsx` would receive the layout defined under `(everyone)`.

Take a visual approach with this tree:

```
❯ tree app/
app/
├── (everyone)
│   ├── layout.tsx
│   ├── login
│   │   └── page.tsx
│   └── page.tsx
└── (users)
    ├── dashboard
    │   └── page.tsx
    └── layout.tsx
```

You see there's a unique landing page at `app/(everyone)/page.tsx`, which uses the layout defined as sibling to it.

Then there's `dashboard`, which because is defined under the users group, uses the layout defined directly under `(users)`.

Careful with accidentally defining two landing pages. If you had a `page.tsx` directly under `(users)` it'd cause a conflict with `(everyone)`'s

**Reply by hamedyosefian** [reply]
Thank you very much, my problem is solved ..  I had another question:
This route had global layout (navbar, sidebar, ...): `/orders/345/`
This route whitout any layout : `/orders/345/invoice/`

how handle this?

**Reply by icyJoseph** [reply]
Something like this:

```
❯ tree app/
app/
├── (detailed)
│   ├── layout.tsx
│   └── orders
│       └── [id]
│           └── invoice
│               └── page.tsx
└── (root)
    ├── layout.tsx
    ├── orders
    │   └── [id]
    │       └── page.tsx
    └── page.tsx
```

**Comment by CaptShmo** [comment]
I have a question about this too. If a client provides me with a specific page they want created with a specific URL, how do I override a root layout without changing the structure of my app? 

Example: Let's say I have a pre-existing page at /deals/weekly that shows weekly deals for an ecommerce site. Then later I get an email from a client requesting a black friday deals page which they want located at /deals/weekly/black-friday, except the design they provide needs a completely new root layout. They say we can't change the URL because the promotion department has already printed it on thousands of flyers they are handing out later in the month. 

Is it possible to make a layout file in a subdirectory / nested route that overrides all the layouts above it in the tree and acts as the route layout for that route?
